src/FavIcon.vala

```diff
@@ -37,14 +37,24 @@ public class FeedReader.FavIcon : GLib.Object
 	private Gee.Promise<Gdk.Pixbuf?> m_pixbuf = null;
 	private ResourceMetadata m_metadata;
 
-	public signal void pixbuf_changed(Feed feed, Gdk.Pixbuf pixbuf);
+	public signal void surface_changed(Feed feed, Cairo.Surface surface);
 
 	private FavIcon(Feed feed)
 	{
 		m_feed = feed;
 	}
 
-	public async Gdk.Pixbuf? get_pixbuf()
+	private int get_scale_factor()
+	{
+		return MainWindow.get_default().get_style_context().get_scale();
+	}
+
+	private Cairo.Surface create_surface_from_pixbuf(Gdk.Pixbuf pixbuf)
+	{
+		return Gdk.cairo_surface_create_from_pixbuf(pixbuf, GetScaleFactor(), null);
```

Comment:
I think you missed `GetScaleFactor` here

Reply:
Indeed I did.